xl-v1-bucket.go

```diff
@@ -63,27 +79,16 @@ func (xl xlObjects) MakeBucket(bucket string) error {
 	// Wait for all make vol to finish.
 	wg.Wait()
 
-	// Look for specific errors and count them to be verified later.
-	for _, err := range dErrs {
-		if err == nil {
-			continue
-		}
-		// if volume already exists, count them.
-		if err == errVolumeExists {
-			volumeExistsErrCnt++
-			continue
-		}
-
-		// Update error counter separately.
-		createVolErr++
+	if !isQuorum(dErrs, xl.writeQuorum) {
+		xl.undoMakeBucket(bucket)
+		return toObjectErr(errXLWriteQuorum, bucket)
 	}
 
-	// Return err if all disks report volume exists.
-	if volumeExistsErrCnt > len(xl.storageDisks)-xl.readQuorum {
-		return toObjectErr(errVolumeExists, bucket)
-	} else if createVolErr > len(xl.storageDisks)-xl.writeQuorum {
-		// Return errXLWriteQuorum if errors were more than allowed write quorum.
-		return toObjectErr(errXLWriteQuorum, bucket)
+	for _, err := range dErrs {
+		if err != nil && err != errDiskNotFound {
+			xl.undoMakeBucket(bucket)
+			return toObjectErr(err, bucket)
+		}
```

Comment:
this has undesired consequence:
1) `mc mb myminio/newbucket` - creates newbucket
2) again do `mc mb myminio/newbucket` - deletes the bucket created in step-1